Kernel/Arch/aarch64/RPi/MMIO.cpp

```diff
@@ -4,17 +4,18 @@
  * SPDX-License-Identifier: BSD-2-Clause
  */
 
+#include <Kernel/Arch/aarch64/CPU.h>
 #include <Kernel/Arch/aarch64/MainIdRegister.h>
 #include <Kernel/Arch/aarch64/RPi/MMIO.h>
 
 namespace Kernel::RPi {
 
 MMIO::MMIO()
-    : m_base_address(0xFE00'0000)
+    : m_base_address(KERNEL_MAPPING_BASE + 0xFE00'0000)
 {
     MainIdRegister id;
     if (id.part_num() <= MainIdRegister::RaspberryPi3)
-        m_base_address = 0x3F00'0000;
+        m_base_address = KERNEL_MAPPING_BASE + 0x3F00'0000;
```

Comment:
This looks strange. These used to be physical addresses, are they not? Are we identity mapping all of physical memory up to 0x2000_0000? or just the kernel image itself? Should we have a separate virtual memory mapping for the MMIO regions?

Reply:
Looking ahead to the next commit, is this one ( Kernel/aarch64: Access MMIO using mapping in high virtual memory ) out of order?

Reply:
At the beginning of the aarch64 port, we used to execute without the MMU, so we could just always access these physical addresses to talk to MMIO. Then, in a62732ee2f029ca388b358e3cc013df9f1951d9d we changed to just identity map the kernel image, and also identity map the MMIO region. This means the code in the Kernel for accessing the MMIO region could just stay the same. Now in this PR, the Kernel image is actually mapped at KERNEL_MAPPING_BASE (0x2000000000), and the MMIO region is moved to KERNEL_MAPPING_BASE + START_OF_MMIO_MEMORY (0x3F000000). So to actually access the MMIO region now we need to access KERNEL_MAPPING_BASE + 0x3F00'0000.

Now I think it would be better if the MemoryManager in the Kernel would actually do the MMIO mapping (as can be done using MemoryManager::allocate_kernel_region), however that would mean there is no debug output before the memory manager is actually initialized.

Reply:
Regarding the order, it's pretty arbitrary, as it was basically impossible to not break booting during these commits. Basically everything should otherwise be one big commit, but I thought it was better to split it up into separate commits. This breaks bisecting but I think that's fine at this stage.

Reply:
Maybe we should only map the serial device, when we initialize the page tables, such that we can actually have some debug output, and then unmap it when the MemoryManager is initialized, and then map the device using the MemoryManager.

Reply:
I updated this commit, and added a NOTE to make it more clear what is happening.